presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -525,30 +524,32 @@ public static LocalTime fromPrestoTime(long value)
             case Types.NUMERIC:
             case Types.DECIMAL:
                 int decimalDigits = type.getRequiredDecimalDigits();
-                int precision = columnSize + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
+                int precision = type.getRequiredColumnSize() + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
                 if (precision > Decimals.MAX_PRECISION) {
                     return Optional.empty();
                 }
                 return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
 
             case Types.CHAR:
             case Types.NCHAR:
-                if (columnSize > CharType.MAX_LENGTH) {
-                    if (columnSize > VarcharType.MAX_LENGTH) {
+                int charLength = type.getRequiredColumnSize();
+                if (charLength > CharType.MAX_LENGTH) {
+                    if (charLength > VarcharType.MAX_LENGTH) {
```

Comment:
Why use both conditions instead of checking `charLength <= CharType.MAX_LENGTH` and otherwise falling through to the `VARCHAR` case?

The char mapping could be modified to use the `Optional` version of `getColumnSize`, but it's not as nice as it is for varchar.

```java
return type.getOptionalColumnSize()
        .filter(length -> length <= CharType.MAX_LENGTH)
        .map(length -> charColumnMapping(createCharType(length)))
        .or(() -> type.getOptionalColumnSize()
                .filter(length -> length <= VarcharType.MAX_LENGTH)
                .map(length -> varcharColumnMapping(createVarcharType(length)))
                .or(() -> Optional.of(varcharColumnMapping(createUnboundedVarcharType()))));
```

Reply:
In case of char, the length must be present. It's erroneous to have a char, but not know its length.